### DISCUSSION TITLE
add virtual columns to s3

### DISCUSSION THREAD

**Discussion by xxx194** [apache/doris#45882] [Ideas]
Hello!
i load 1000 files by mask *  

  select 
_path ,
*

FROM s3(
        "URI" = "s3://xxx/yyy/*.snappy.parquet",
        "resource" = "<>",
        "format" = "parquet"
    ) t

Virtual columns
_path — Path to the file. Type: LowCardinalty(String).
_file — Name of the file. Type: LowCardinalty(String).
_size — Size of the file in bytes. Type: Nullable(UInt64). If the size is unknown, the value is NULL.
_time — Last modified time of the file. Type: Nullable(DateTime). If the time is unknown, the value is NULL.
_etag — ETag of the file. Type: LowCardinalty(String). If the etag is unknown, the value is NULL.

**Comment by ixzc** [comment]
what's the meaning of you? can you provide an example?

**Reply by xxx194** [reply]
![image](https://github.com/user-attachments/assets/72add098-9ddd-4a44-844b-720cf58eaa6d)

and for kafka add metadata columns like 

COLUMNS (
    topic, partition, timestamp, key, value, headers
);
